torch/csrc/api/src/nn/modules/normalization.cpp

```diff
@@ -0,0 +1,40 @@
+#include <torch/nn/modules/normalization.h>
+
+#include <torch/cuda.h>
+#include <torch/types.h>
+#include <torch/utils.h>
+#include <torch/nn/init.h>
+
+#include <ostream>
+#include <utility>
+
+namespace torch {
+namespace nn {
+
+LayerNormImpl::LayerNormImpl(const LayerNormOptions& options_) : options(options_) {
+  reset();
+}
+
+void LayerNormImpl::reset() {
+  if (options.elementwise_affine()) {
+    weight = register_parameter("weight", torch::ones(torch::IntArrayRef(options.normalized_shape())));
+    bias = register_parameter("bias", torch::zeros(torch::IntArrayRef(options.normalized_shape())));
+  } else {
+    weight = register_parameter("weight", torch::empty(torch::IntArrayRef(options.normalized_shape())));
+    bias = register_parameter("bias", torch::empty(torch::IntArrayRef(options.normalized_shape())));
```

Comment:
This if-branch maps to Python version:
```python
self.register_parameter('weight', None)
self.register_parameter('bias', None)
```
to achieve parity, the corresponding C++ version should be:
```cpp
weight = register_parameter("weight", torch::Tensor(), /*requires_grad=*/false);
bias = register_parameter("bias", torch::Tensor(), /*requires_grad=*/false);
```

Reply:
alright